nand/nand_ftl_diskio.c

```diff
@@ -66,7 +68,7 @@ DRESULT diskio_read(BYTE drv, BYTE *buff, LBA_t sector, UINT count)
     dhara_error_t err;
 
 	if (drv) return STA_NOINIT;		/* Supports only drive 0 */
-
+    mutex_enter_blocking(&diskio_mutex);
```

Comment:
Can you help me understand what this mutex is protecting against?

I'm trying to see where both CPU cores would enter a state that would cause a conflict.  I'm sure you found something, and want to ensure it's documented, so future code changes don't inadvertently break stuff again through (likely my) ignorance.

Reply:
Dhara isn't meant for concurrent access (not thread safe by design). Core1 is handling USB thus all USBMS operations. Core0 is doing the local file operations. 
It is possible for Core1 to do a write while Core0 is doing a read. The mutex is to protect against this kind of situation.

Reply:
Nice catch!